src/main/java/edu/wright/airviewer2/AIRViewerController.java

```diff
@@ -392,6 +397,40 @@ public void handle(ActionEvent e) {
 		refreshUserInterface();
 		return model;
 	}
+	//Initializing Sign menu item
+	private void initSignMenu() {
+
+        signMenuItem.setOnAction(e -> signDocument());
+    }
+	
+	private void signDocument() {
+        // Create a Page object
+        PDPage pdPage = new PDPage();
+        // Add the page to the document and save the document to a desired file.
+        model.wrappedDocument.addPage(pdPage);
+
+        try {
+
+            PDSignature pdSignature = new PDSignature();
+            pdSignature.setFilter(PDSignature.FILTER_VERISIGN_PPKVS);
+            pdSignature.setSubFilter(PDSignature.SUBFILTER_ADBE_PKCS7_SHA1);
+
+            pdSignature.setName("AirViewer Crew");
+            pdSignature.setLocation("WFH");
+            pdSignature.setReason("Signature Validation");
+            pdSignature.setSignDate(Calendar.getInstance());
+            model.wrappedDocument.addSignature(pdSignature, null);
+
+            model.wrappedDocument.save(path);
+            MessageBox.show("Added Signature successfully", "Alert");
+
+        } catch (IOException ioe) {
+            System.out.println("Error while saving pdf. Please try again later" + ioe.getMessage());
+            MessageBox.show(ioe.toString(), "Io Exception");
```

Comment:
add more readable message to the UI than actual exceptions. add the the exception details in the log.